lib/repl.js

```diff
@@ -679,23 +678,19 @@ function REPLServer(prompt,
         if (typeof stackFrames === 'object') {
           // Search from the bottom of the call stack to
           // find the first frame with a null function name
-          const idx = ArrayPrototypeFindIndex(
-            ArrayPrototypeReverse(stackFrames),
+          const idx = ArrayPrototypeFindLastIndex(
+            stackFrames,
             (frame) => frame.getFunctionName() === null,
           );
           // If found, get rid of it and everything below it
-          frames = ArrayPrototypeSplice(stackFrames, idx + 1);
+          frames = ArrayPrototypeSlice(stackFrames, 0, idx);
         } else {
           frames = stackFrames;
         }
-        // FIXME(devsnek): this is inconsistent with the checks
-        // that the real prepareStackTrace dispatch uses in
-        // lib/internal/errors.js.
         if (typeof Error.prepareStackTrace === 'function') {
           return Error.prepareStackTrace(error, frames);
         }
```

Comment:
I think to really address the FIXME what you need is replacing this with:

```suggestion
        const globalOverride = maybeOverridePrepareStackTrace(globalThis, error, frames);
        if (globalOverride !== kNoOverride) return globalOverride;
```

...in case someone deletes `globalThis.Error` (this is the REPL, where people can do crazy things, after all..)

Reply:
Thank you for pointing this out. I think in the override function there is no handy way to get the globalThis of the REPL context. I'll fix this FIXME in a follow-up.